src/DiagnosticMessages.ts

```diff
@@ -24,711 +28,841 @@ export let DiagnosticMessages = {
+        code: 'expected-in-for-each'
     }),
     expectedExpressionAfterForEachIn: () => ({
         message: `Expected expression after 'in' keyword from 'for each' statement`,
-        code: 1053,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1053,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-expression-for-each'
     }),
     unexpectedColonBeforeIfStatement: () => ({
         message: `Colon before 'if' statement is not allowed`,
-        code: 1054,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1054,
+        severity: DiagnosticSeverity.Error,
+        code: 'unexpected-colon-before-if'
     }),
     expectedStringLiteralAfterKeyword: (keyword: string) => ({
         message: `Missing string literal after '${keyword}' keyword`,
-        code: 1055,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1055,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-string-literal'
     }),
     keywordMustBeDeclaredAtRootLevel: (keyword: string) => ({
         message: `${keyword} must be declared at the root level`,
-        code: 1056,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1056,
+        severity: DiagnosticSeverity.Error,
+        code: 'keyword-must-be-root-level'
     }),
     __unused5: () => ({
         message: `'library' statement must be declared at the top of the file`,
-        code: 1057,
+        legacyCode: 1057,
         severity: DiagnosticSeverity.Error
     }),
     expectedEndIfElseIfOrElseToTerminateThenBlock: () => ({
         message: `Expected 'end if', 'else if', or 'else' to terminate 'then' block`,
-        code: 1058,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1058,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-terminator-on-then'
     }),
     expectedEndTryToTerminateTryCatch: () => ({
         message: `Expected 'end try' to terminate 'try-catch' statement`,
-        code: 1059,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1059,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-terminator-on-try-catch'
     }),
     expectedEndIfToCloseIfStatement: (startingPosition: Position) => ({
         message: `Expected 'end if' to close 'if' statement started at ${startingPosition?.line + 1}:${startingPosition?.character + 1}`,
-        code: 1060,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1060,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-terminator-on-if'
     }),
     expectedStatementToFollowConditionalCondition: (conditionType: string) => ({
         message: `Expected a statement to follow '${conditionType?.toLowerCase()} ...condition... then'`,
-        code: 1061,
-        severity: DiagnosticSeverity.Error
+        legacyCode: 1061,
+        severity: DiagnosticSeverity.Error,
+        code: 'expected-statement-after-conditional'
```

Comment:
Maybe merge a few of these together, parameterize the message a bit.

Reply:
consolidated all the `expected-statement-*` diagnostics